### DISCUSSION TITLE
Compiling on Linux

### DISCUSSION THREAD

**Discussion by jamesarbrown** [euzu/tuliprox#67] [Q&A] [answered]
Hi,

Can you help advise the process for compiling on linux (x86 Alpine)

I can get bin/build_lin.sh to work on its own, but then I do not know where to populate the content that is needed in /web, so it runs without the UI working

I think release.sh is the entry point, installed yarn etc, but Busybox is not happy and bounces a grep issue on startup.

```
/opt/m3u-filter# ./bin/release.sh
grep: unrecognized option: P
BusyBox v1.36.1 (2024-06-10 07:11:47 UTC) multi-call binary.
```

I guess I could run containerd on here, but I was trying to keep it extremely light and run it all in Alpine LXC. It seems heavy to sit Docker into it.

TIA

**Comment by euzu** [comment]
@jamesarbrown hi,

i am on vacation and dont have access to my pc, 

I am compiling the fe with node and yarn.
I have node and yarn installed on linux.
```shell
cd ./frontend
yarn
yarn build
````
The build directory contains the web files.



If you like docker, this could work with docker  to build the frontend.

This is the Dockerfile

```dockerfile
# Stage 1: Build the React app
FROM node:lts as build-stage

ENV NODE_OPTIONS=--openssl-legacy-provider

WORKDIR /app
COPY ./frontend /app

RUN yarn install \
  --prefer-offline \
  --frozen-lockfile \
  --non-interactive \
  --production=false

RUN yarn build

# Stage 2: Copy the build output to a new image
FROM node:lts as production-stage

WORKDIR /app
COPY --from=build-stage /app/build ./build

# This stage is just to output the build directory to the local filesystem
FROM scratch as export-stage
COPY --from=production-stage /app/build /build
```

```shell
# build docker  image
docker build -t m3u-fe .
# create container 
docker create --name m3u-fe-container m3u-fe
# copy build to local disk
docker cp m3u-fe-container:/build ./web
# delete container
docker rm m3u-fe-container
```

**Comment by jamesarbrown** [comment, marked as answer]
Hi,
Thanks for that. I dont know if any use in the documentation, but here are the commands to get it started in a Alpine 3.19 LXC

**Installing in LXC Container (Alpine)**
```
apk update
apk add nano git yarn bash cargo perl-local-lib perl-module-build make 
cd /opt
git clone https://github.com/euzu/m3u-filter.git
cd /opt/m3u-filter/bin
./build_lin.sh
ln -s /opt/m3u-filter/target/release/m3u-filter /bin/m3u-filter 
cd /opt/m3u-filter/frontend
yarn
yarn build
ln -s /opt/m3u-filter/frontend/build /web
ln -s /opt/m3u-filter/config /config
mkdir /data
mkdir /backup
```

**Creating a service, create /etc/init.d/m3u-filter**
```
#!/sbin/openrc-run
name=m3u-filter
command="/bin/m3u-filter"
command_args="-p /config -s"
command_user="root"
command_background="yes"
output_log="/var/log/m3u-filter/m3u-filter.log"
error_log="/var/log/m3u-filter/m3u-filter.log"
supervisor="supervise-daemon"

depend() {
    need net
}

start_pre() {
    checkpath --directory --owner $command_user:$command_user --mode 0775 \
           /run/m3u-filter /var/log/m3u-filter
}
```

**then add it to boot**
```rc-update add m3u-filter default```

**Comment by euzu** [comment]
[Wiki](https://github.com/euzu/m3u-filter/wiki/alpinelxc)

Thank you for your description. I have added it to the WIKI
